### DISCUSSION TITLE
How to give a speaker name to an LJSpeech dataset ?

### DISCUSSION THREAD

**Discussion by Ca-ressemble-a-du-fake** [coqui-ai/TTS#1180] [General Q&A]
Hi,

I've got my first fine-tuned model trained on Exp3 (En, pt, fr) VITS model and when listing speaker idx included in the model, I found that speaker name was "ljspeech". I would prefer to have the name of the voice I trained the model on, because I aim at having other voices.
THe command I input to list the speakers was : 
`
!tts --model_path "{modelDir}{checkpointName}" --config_path "{modelDir}config.json" --list_speaker_idxs
`
And the end of the output was :

> (Set --speaker_idx flag to one of these values to use the multi-speaker model.
{'ED': 0, 'VCTK_p225': 1, 'VCTK_p226': 2,, 'bernard': 109, 'ezwa': 110, 'gilles_g_le_blanc': 111, **'ljspeech': 112,** 'nadine_eckert_boulet': 113, 'zeckou': 114}

I saw in dataset class that there was an argument `speaker_id_mapping` but it looks like it is for d-vector. I could not find where the returned speaker_idx defaults to the dataset name.

So do you know how to give a speaker name to an LJSpeech dataset ?

Thanks

**Comment by Ca-ressemble-a-du-fake** [comment]
I've just found in [formatter](https://github.com/coqui-ai/TTS/blob/c63bb481e95bd4a1ff978947d8e3e6c0bfb4177f/TTS/tts/datasets/formatters.py#L116) class that speaker name was always given the name "ljspeech".

**Reply by jaggukaka** [reply]
I think ljspeech has only one speaker, so you can just use that name. Generally if you have custom dataformat, your formatter should be able to handle this in right way. Your custom formatter should return List[[text, wav_file, speaker]], so whatever the formatter returns in the last index of the inner list, your speaker should get the same identifier. If you are using existing formatter, make sure you arrange your dataset dirs and files in such a way the formatter expects, and see if the output of formatter matches with what you expect.

**Reply by Ca-ressemble-a-du-fake** [reply]
Great, thank you @jaggukaka, I'll make some tests to know what the formatter expects!
